cmd/scanner/scanner.go

```diff
@@ -53,14 +74,13 @@ func getProvider(provider string) inventory.CloudProvider {
 	return inventory.UnknownProvider
 }
 
-// getCloudProviderAccounts retrieves cloud provider accounts from a credentials file.
-func getCloudProviderAccounts() ([]inventory.Account, error) {
-	accounts := make([]inventory.Account, 0)
-
+// readCloudProviderAccounts reads and loads cloud provider accounts from a credentials file.
+func (s *Scanner) readCloudProviderAccounts() error {
 	// Load cloud accounts credentials file.
-	cfg, err := ini.Load(credsFile)
+	cfg, err := ini.Load(s.credsFile)
 	if err != nil {
-		return nil, fmt.Errorf("%w", err)
+		s.logger.Panic("Can't load credentials file. Aborting...", zap.Error(err))
```

Comment:
1. I don't think we need to panic here, we need to return the error to the caller
2. I think that the calling function should log the error, otherwise there will be a duplicate in the logs
Please take a look [here](https://github.com/uber-go/guide/blob/master/style.md#handle-errors-once)
```text
{"level":"error","timestamp":"2023-09-11T09:02:22.277+0300","msg":"Can't load credentials file. Aborting...","error":"open /home/vbelouso/repo/projects/cluster-iq/secrets/credentials: no such file or directory"}
{"level":"error","timestamp":"2023-09-11T09:02:22.277+0300","msg":"Failed to get cloud provider accounts","error":"open /home/vbelouso/repo/projects/cluster-iq/secrets/credentials: no such file or directory"}
```

Reply:
Good feedback, Done